core/client/src/main/java/alluxio/client/file/FileOutStream.java

```diff
@@ -198,21 +205,12 @@ public void close() throws IOException {
         if (mCanceled) {
           // TODO(yupeng): Handle this special case in under storage integrations.
           mUnderStorageOutputStream.close();
-          if (!ufs.exists(tmpPath)) {
-            // Location of the temporary file has changed, recompute it.
-            updateUfsPath();
-            tmpPath = PathUtils.temporaryFileName(mNonce, mUfsPath);
-          }
           ufs.delete(tmpPath, false);
         } else {
           mUnderStorageOutputStream.flush();
           mUnderStorageOutputStream.close();
-          if (!ufs.exists(tmpPath)) {
-            // Location of the temporary file has changed, recompute it.
-            updateUfsPath();
-            tmpPath = PathUtils.temporaryFileName(mNonce, mUfsPath);
-          }
-          if (!ufs.rename(tmpPath, mUfsPath)) {
+          if (!ufs.rename(tmpPath, mUfsPath)) { // Failed to commit file
+            ufs.delete(tmpPath, false);
```

Comment:
Yeah I was trying to make the behavior consistent with when delegation is enabled. Do you think we should not delete the temporary file? While it may be nice to be able to manually recover, if the write operation is programmatic, it can possible create lots of unmanaged temporary files.